Diplom_2

avonurumanna opened the pull request.

avonurumanna committed:
add changed pom

avonurumanna committed:
add clients

avonurumanna committed:
add generator

avonurumanna committed:
add pojo

avonurumanna committed:
add tests

avonurumanna committed:
add report

pavel-ulan commented on src/main/java/pojo/Data.java:
```diff
@@ -0,0 +1,137 @@
+import com.google.gson.annotations.SerializedName;
+
+public class Data {
+    @SerializedName("__v")
```
**отлично**
молодец, что пользуешься мапингом

pavel-ulan commented on src/test/java/CreateUserTest.java:
```diff
@@ -0,0 +1,82 @@
+                .and()
+                .assertThat().body("refreshToken", notNullValue());
+
+        token = createUserResponse.extract().path("accessToken");
```
**можно лучше**
подход верный, но есть одно упущение. Так если тест по какой-то причине не пройдет asserts, то эта строка не выполнится (соответственно не будет токена и не произойдет удаления пользователя). Такое не должно быть, если пользователь был создан, он должен быть удален в любом случае.

pavel-ulan commented on src/test/java/ParameterizedEditUserTest.java:
```diff
@@ -0,0 +1,77 @@
+
+    @Test
+    @DisplayName("Check that unauthorized user can not be updated")
+    public void unauthorizedUserCanNotBeUpdated() {
```
**можно лучше**
старайся использовать в названии тестовых методов шаблон ***Test (например не unauthorizedUserCanNotBeUpdated, а unauthorizedUserCanNotBeUpdatedTest). это влияет на работу некоторых систем отчета и сторонних библиотек.

pavel-ulan reviewed (approved):
Отлично. Я оставил пару комментов для информации, посмотри их.

avonurumanna committed:
changes according to the reviewer comments:
names of tests were changes according to format ***Test
saving of token was moved to step after creating user
